tools/api-markdown-documenter/src/utilities/ApiItemUtilities.ts

```diff
@@ -171,29 +171,54 @@ export function getReleaseTag(apiItem: ApiItem): ReleaseTag | undefined {
 }
 
 /**
- * Creates a string representation of the provided release tag.
+ * Gets all {@link https://tsdoc.org/pages/spec/tag_kinds/#modifier-tags | modifier tags} associated with the provided API item.
  *
- * @remarks If `None`, this will return an empty string.
+ * @remarks Note that this will include both standard and any preserved custom tags.
+ *
+ * @public
  */
-export function releaseTagToString(releaseTag: ReleaseTag): string {
-	// eslint-disable-next-line default-case
-	switch (releaseTag) {
-		case ReleaseTag.Alpha: {
-			return "Alpha";
-		}
-		case ReleaseTag.Beta: {
-			return "Beta";
-		}
-		case ReleaseTag.Internal: {
-			return "Internal";
-		}
-		case ReleaseTag.Public: {
-			return "Public";
+export function getModifierTags(apiItem: ApiItem): ReadonlySet<string> {
+	const modifierTags = new Set<string>();
+	if (apiItem instanceof ApiDocumentedItem && apiItem.tsdocComment !== undefined) {
+		for (const tag of apiItem.tsdocComment.modifierTagSet.nodes) {
+			modifierTags.add(tag.tagName);
 		}
-		case ReleaseTag.None: {
-			return "";
+	}
+	return modifierTags;
+}
+
+/**
+ * Checks if the provided API item is tagged with the specified {@link https://tsdoc.org/pages/spec/tag_kinds/#modifier-tags | modifier tag}.
+ *
+ * @throws If the provided tag name is malformed (i.e., does not begin with `@`).
+ *
+ * @public
+ */
+export function hasModifierTag(apiItem: ApiItem, tagName: string): boolean {
+	if (!tagName.startsWith("@")) {
+		throw new Error("Invalid TSDoc tag name. Tag names must start with `@`.");
+	}
+	return getModifierTags(apiItem).has(tagName);
+}
+
+/**
+ * Gets all custom {@link https://tsdoc.org/pages/spec/tag_kinds/#block-tags | block comments} associated with the provided API item.
+ * @returns A mapping from tag name to the associated block contents.
+ *
+ * @public
+ */
+export function getCustomBlockComments(apiItem: ApiItem): ReadonlyMap<string, DocSection[]> {
+	const customBlockComments = new Map<string, DocSection[]>();
+	if (apiItem instanceof ApiDocumentedItem && apiItem.tsdocComment?.customBlocks !== undefined) {
+		for (const block of apiItem.tsdocComment.customBlocks) {
+			if (!customBlockComments.has(block.blockTag.tagName)) {
+				customBlockComments.set(block.blockTag.tagName, []);
+			}
+			// eslint-disable-next-line @typescript-eslint/no-non-null-assertion -- Guaranteed to exist because we added it above
+			customBlockComments.get(block.blockTag.tagName)!.push(block.content);
```

Comment:
This pattern for doing default initialization in maps is pretty inefficient since it hashes the key and does the lookup 2-3 times and annoying (as it fails linting).

Something like the pattern in getOrAddEmptyToMap in /packages/dds/tree/src/util/utils.ts avoids the need to assert with "!" and saves a map lookup by introducing a variable: I think that might be a better pattern.